extensions/objects/templates/ListOfSetsOfHtmlStringsEditor.js

```diff
@@ -0,0 +1,104 @@
+// Copyright 2018 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+oppia.directive('listOfSetsOfHtmlStringsEditor', [
+  '$compile', 'OBJECT_EDITOR_URL_PREFIX',
+  function($compile, OBJECT_EDITOR_URL_PREFIX) {
+    return {
+      link: function(scope, element) {
+        scope.getTemplateUrl = function() {
+          return OBJECT_EDITOR_URL_PREFIX + 'ListOfSetsOfHtmlStrings';
+        };
+        $compile(element.contents())(scope);
+      },
+      restrict: 'E',
+      scope: {
+        getInitArgs: '&',
+        value: '='
+      },
+      template: '<span ng-include="getTemplateUrl()"></span>',
+      controller: ['$scope', function($scope) {
+        var errorMessage = '';
+
+        if (!$scope.value) {
+          $scope.value = [];
+        }
+        if (!$scope.maxPrevIndex) {
+          $scope.maxPrevIndex = 0;
+        }
+        $scope.initArgs = $scope.getInitArgs();
+        $scope.choices = $scope.initArgs.choices;
+
+        if ($scope.selectedRank !== '' && $scope.selectedRank !== null) {
+          if ($scope.maxPrevIndex < parseInt($scope.selectedRank)) {
+            $scope.maxPrevIndex = parseInt($scope.selectedRank);
+          }
+        }
+
+        $scope.allowedChoices = function() {
+          var allowedList = [];
+          for (var i = 1; i <= $scope.maxPrevIndex + 1; i++) {
```

Comment:
Why not start at position 0?

Reply:
These ranks corresponds to the allowable rank shown to the creator. I think we'll prefer the creator to start from 1 rather than 0, right?

Reply:
Done